File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
    let dummy= new ListNode(-1)
    let temp = dummy
    while(l1!== null || l2!= null){
        if(l1 && l2){
            
            if(l2.val <= l1.val){
                temp.next = new ListNode(l2.val) 
                temp= temp.next
                l2 = l2.next
            }
            else{
                temp.next = new ListNode(l1.val) 
                temp=temp.next
                l1 = l1.next
            }
        }
        else break
    }
    if(l1&& !l2){
         temp.next = l1
    }
    if(!l1 && l2){
       temp.next = l2
    }
    return dummy.next
};

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1)
        temp = dummy
        while(l1 != None and l2!= None):       
            if(l2.val <= l1.val):
                temp.next =  ListNode(l2.val) 
                temp= temp.next
                l2 = l2.next
            else:
                temp.next = ListNode(l1.val) 
                temp=temp.next
                l1 = l1.next
        if(l1!= None and l2 == None):
             temp.next = l1
        if(l1 == None and l2 != None):
            temp.next = l2
        return dummy.next